### PR TITLE
Add meta and OpInfo for _embedding_bag_dense_backward

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #109211

The sample inputs is a bit involved because there are a lot of
shenanigans in the derivative formula.  Check comments.

This is exercised in vdd, internal test `buck2 run '@fbcode//mode/opt' fbcode//pytorch/benchmark/fb/test_gpu:run_test_gpu -- 'pytorch.benchmark.fb.test_gpu.test_gpu.TestBenchmarkFbGpu.test_train_blue_reels_vdd_v3_inductor_speedup'`

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

### CODE REVIEW

File: torch/testing/_internal/common_methods_invocations.py [hunk 1]
```diff
@@ -7290,6 +7290,69 @@ def sample_inputs_dropout_backward(op_info, device, dtype, requires_grad, **kwar
...
+        if padding_idx is None:
+            padding_idx = -1
+        else:
+            from torch._meta_registrations import maybe_wrap_dim
+            padding_idx = maybe_wrap_dim(padding_idx, weight.size(0))
+
+        sample.kwargs.pop('max_norm', None)  # not used by the op
+        sample.kwargs.pop('norm_type', None)  # not used by the op
+
+        # This is a precondition for the function, so force it
```
Comment (albanD) [inline torch/testing/_internal/common_methods_invocations.py:7327]: Isn't this already True if it is a precondition for the forward?

**Review by albanD — APPROVED** [review]
SGTM

File: torch/testing/_internal/common_methods_invocations.py (hunk 1 above)
Comment (ezyang) [reply torch/testing/_internal/common_methods_invocations.py:7327]: No, not precondition for forward. The backward only gets called in contexts where the internal private function was guaranteed to be contiguous

Comment (albanD) [reply torch/testing/_internal/common_methods_invocations.py:7327]: Ok ok

**Review by zou3519 — APPROVED** [review]

**Pull request closed without merging**